### ISSUE TITLE
Implement a copy operator

### ISSUE THREAD

**Issue by Dandigit** [enact-lang/enact#61]
I've decided to move away from the references proposal described in #60, and instead pass Objects by reference by default.

To replace the added control that references would have added, I've decided to add a new `copy` operator, which acts like this:
```
struct Person:
    name string
    age int
end

fun setAgeTo99(people [Person]) int:
    each person in people:
        person.age = 99
    end
end

var people = [Person("Jim", 20), Person("Sam", 84), Person("Matt", 42)]

setAgeTo99(copy people)
// `people` in this scope is unaffected

setAgeTo99(people)
// `people` in this scope are now all 99
```

**Comment by Dandigit** [comment]
Decided not to implement a `move` operator. It adds more complexity and runtime overhead without much real benefit.

**Comment by Dandigit** [comment]
Okay, after more in-depth consideration, I think I will move back to the proposal in #60, with a couple of other changes to make it fit with the language better.

Hopefully this is final.

**Comment by Dandigit** [comment]
Implemented in #62.
